HW27/src/main/java/org/example/order/CoffeeOrderBoard.java

```diff
@@ -0,0 +1,76 @@
+package org.example.order;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+
+public class CoffeeOrderBoard {
+
+    private static final Logger log = LoggerFactory.getLogger(CoffeeOrderBoard.class);
+
+    ArrayDeque<Order> deque;
+    private int lastOrderNumber;
+
+    public CoffeeOrderBoard() {
+        deque = new ArrayDeque<>();
+    }
+
+    public void add(String clientName) {
+        deque.addLast(new Order(clientName, ++lastOrderNumber));
+
+        log.info("Added client name: " + clientName);
+    }
+
+    public String deliver() {
+        String result;
+        Order order;
+        order = deque.pollFirst();
+        if (order != null) {
+            result = "order from " + order.getClientName() + " number " + order.getOrderNumber() + " has been deleted";
+        } else {
+            result = "Order list is empty";
+        }
+
+        log.info("Deleted order: " + order);
+
+        return result;
+    }
+
+    public String deliver(int orderNumber) {
+
+        String result;
+        Order order = new Order(null, 0);
+        try {
+            for (Order o : this.deque) {
+                if (o.getOrderNumber() == orderNumber) {
+                    order = o;
+                }
+            }
+            if (this.deque.remove(order)) {
+                result = "order deleted " + order.getClientName() + " " + order.getOrderNumber();
+
+                log.info("Deleted order: " + order);
+            } else {
+                throw new Exception("Order" + orderNumber + " is missing");
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage() + " " + Arrays.toString(e.getStackTrace()));
```

Comment:
in logger better to avoid + and use {} for placcing
log.error("Error apeared: {} {}", e.getMessage(), Arrays.toString(e.getStackTrace()));

Arrays.toString(e.getStackTrace()) could be transferred just to e

log.error("Error apeared: {} {}", e.getMessage(), e);